packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -128,13 +128,17 @@ const devHtmlHook: IndexHtmlTransformHook = async (
     if (module) {
       server?.moduleGraph.invalidateModule(module)
     }
-
-    s.overwrite(
-      node.loc.start.offset,
-      node.loc.end.offset,
-      `<script type="module" src="${modulePath}"></script>`,
-      { contentOnly: true }
-    )
+    if (ext === 'js') {
+      s.overwrite(
+        node.loc.start.offset,
+        node.loc.end.offset,
+        `<script type="module" src="${modulePath}"></script>`,
+        { contentOnly: true }
+      )
+    } else if (ext === 'css') {
+      // just use the style update hmr don't render css
+      s.append(`<script type="module" src="${modulePath}"></script>`)
+    }
```

Comment:
When I try the assets playground and I edit the color in css/import.css, I get a full page reload. Is that expected?

Regardless, it is quite difficult to reason about what is happening now because in the inline `<style>` remains in the HTML and there is a new script tag with the proxy CSS injected. So we have the same style twice. 

Ping @bluwy @sapphi-red, this PR seems important, in case you have some time to review (there is a regression linked). Maybe I'm missing something here.

Reply:
> When I try the assets playground and I edit the color in css/import.css, I get a full page reload. Is that expected?

https://github.com/vitejs/vite/blob/c481f26892e5da678d0232a86cbc79fa69ffc674/packages/playground/assets/index.html#L230-L232

Yes, PR will make the connection between CSS (including nested CSS) and Html and if HTML update will `full-reload`.
when `index.html` update will full-reload.
when nested CSS update also full-reload. it will make the index.html change.


But there is still a little problem in ssr. I deliberately removed the # of 6859. 🥲

```js
s.append(`<script type="module" src="${modulePath}"></script>`)
```

This is using the script request the css plugin handle the deps on inline css.